feature_engine/variable_manipulation.py

```diff
@@ -93,6 +93,20 @@ def _find_or_check_numerical_variables(
     return variables
 
 
+def contains_categorical(column: pd.Series) -> bool:
+    return is_object(column) and (
```

Comment:
instead of having this massive return statement, could we include the logic inside the function to output a value? and then return that value?

This logic is designed to tackle the 2 situations you very clearly explained in the PR. The thing is, soon after we merge this PR, we will forget (or at least I will). So could we capture some of the reasoning / problematic cases in comments before the logic?

There are 2 situations, one if it is categorical, maybe capture that individually in some logic (with an if loop?). And the second one if it is object, maybe capture that in a second part of a loop or something. Both blocks of code could output a value. And then, the function returns that value.

Could you see if something on those lines would be possible? it will help us maintain.